### PR TITLE
Refactor Secret Management API

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
## Purpose
$subject

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/secret/management/v1/SecretsApi.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
     @Path("/{secret-type}")
     @Consumes({"application/json"})
     @Produces({"application/json"})
-    @ApiOperation(value = "Create a secret", notes = "This API provides the capability to create a secret ", response = SecretResponse.class, authorizations = {
+    @ApiOperation(value = "Create a secret", notes = "This API provides the capability to create a secret. ", response = SecretResponse.class, authorizations = {
```
Comment (ashendes) [inline components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/secret/management/v1/SecretsApi.java:46]: Remove extra space after the period.

File: components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java [hunk 2]
```diff
@@ -1,5 +1,4 @@
 /*
- * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ashendes) [inline components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java:2]: This is the correct line. Please add this and remove the next line instead.

File: components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java [hunk 3]
```diff
@@ -179,8 +182,9 @@ public SecretResponse getSecret(String secretType, String name) {
     /**
      * To make a partial update or update the specific property of the secret.
      *
-     * @param name               secret name.
-     * @param secretPatchRequest secret's patch details.
+     * @param secretType Secret type name.
```
Comment (ashendes) [inline components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java:185]: Description indentation not consistent with the params/descriptions formatting below.

File: components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java [hunk 4]
```diff
@@ -223,8 +227,9 @@ public SecretResponse patchSecret(String secretType, String name, SecretPatchReq
     /**
      * Update secret details by name.
      *
-     * @param name                secret name.
-     * @param secretUpdateRequest secret's updated details.
+     * @param secretType Secret type name.
```
Comment (ashendes) [inline components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java:230]: Inconsistent formatting.

File: components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/secret/management/v1/SecretsApi.java (hunk 1 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/gen/java/org/wso2/carbon/identity/api/server/secret/management/v1/SecretsApi.java]: Resolved

File: components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java (hunk 4 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/java/org/wso2/carbon/identity/api/server/secret/management/v1/core/SecretManagementService.java]: Resolved

**Review by ashendes — APPROVED** [review]

**Review by ashendes — APPROVED** [review]

**Pull request merged**